geodata/models/raster/reader.py

```diff
@@ -2,17 +2,55 @@
 from celery.utils.log import get_task_logger
 from django.contrib.gis.gdal import SpatialReference
 from django.contrib.gis.geos import Polygon
+import numpy as np
 from osgeo import gdal
 import rasterio
 
 from rgd.utility import _field_file_to_local_path
 from .base import BandMetaEntry, ConvertedRasterFile, RasterEntry
 from .ifiles import RasterFile
 from ..common import _ReaderRoutine
+from ..geometry.transform import transform_geometry
 
 logger = get_task_logger(__name__)
 
 
+def convex_hull(points):
+    from scipy.spatial import ConvexHull
+
+    hull = ConvexHull(points)
+
+    boundary = points[hull.vertices]
+    # Close the loop
+    boundary = np.append(boundary, boundary[0][None], axis=0)
+
+    return boundary
+
+
+def get_valid_data_footprint(src, band_num):
+    """Fetch points for the footprint polygon of the valid data.
+
+    Must specify band of the raster to evaluate.
+
+    src is an open dataset with rasterio
+
+    Returns a numpy array of the bounadry points in a closed polygon.
+
+    """
+    msk = src.read_masks(band_num)
+
+    a = (np.arange(msk.shape[1]) * src.res[1]) + src.bounds.left
+    b = (np.arange(msk.shape[0]) * src.res[0]) + src.bounds.bottom
+    xx, yy = np.meshgrid(a, b[::-1])
```

Comment:
These three lines also make an assumption about the coordinate system. We may want to wait until after the transform to do this

Reply:
Also, this is using the corner of a pixel as representative of the point.  Pedantically, this means right and top edges will be excluded from the returned convex hull.

Reply:
@manthey, I've updated this to compute the coordinates at all cell centers. This means there are XY coordinates for every cell in the mask array rather than on the lower-left edges of all cells

Reply:
Actaully, I messed this up. Will fix again

Reply:
Okay, now it is fixed